src/components/Videos.vue

```diff
@@ -121,6 +139,227 @@ export default {
+					ff: 'Fulah',
+					fi: 'Finnish',
+					fj: 'Fijian',
+					fo: 'Faroese',
+					fr: 'French',
+					fy: 'Western Frisian',
+					ga: 'Irish',
+					gd: 'Gaelic',
+					gl: 'Galician',
+					gn: 'Guarani',
+					gu: 'Gujarati',
+					gv: 'Manx',
+					ha: 'Hausa',
+					he: 'Hebrew',
+					hi: 'Hindi',
+					ho: 'Hiri Motu',
+					hr: 'Croatian',
+					ht: 'Haitian',
+					hu: 'Hungarian',
+					hy: 'Armenian',
+					hz: 'Herero',
+					ia: 'Interlingua',
+					id: 'Indonesian',
+					ie: 'Interlingue',
+					ig: 'Igbo',
+					ii: 'Sichuan Yi',
+					ik: 'Inupiaq',
+					io: 'Ido',
+					is: 'Icelandic',
+					it: 'Italian',
+					iu: 'Inuktitut',
+					ja: 'Japanese',
+					jv: 'Javanese',
+					ka: 'Georgian',
+					kg: 'Kongo',
+					ki: 'Kikuyu',
+					kj: 'Kuanyama',
+					kk: 'Kazakh',
+					kl: 'Kalaallisut',
+					km: 'Central Khmer',
+					kn: 'Kannada',
+					ko: 'Korean',
+					kr: 'Kanuri',
+					ks: 'Kashmiri',
+					ku: 'Kurdish',
+					kv: 'Komi',
+					kw: 'Cornish',
+					ky: 'Kirghiz',
+					la: 'Latin',
+					lb: 'Luxembourgish',
+					lg: 'Ganda',
+					li: 'Limburgan',
+					ln: 'Lingala',
+					lo: 'Lao',
+					lt: 'Lithuanian',
+					lu: 'Luba-Katanga',
+					lv: 'Latvian',
+					mg: 'Malagasy',
+					mh: 'Marshallese',
+					mi: 'Maori',
+					mk: 'Macedonian',
+					ml: 'Malayalam',
+					mn: 'Mongolian',
+					mr: 'Marathi',
+					ms: 'Malay',
+					mt: 'Maltese',
+					my: 'Burmese',
+					na: 'Nauru',
+					nb: 'Norwegian Bokmål',
+					nd: 'Ndebele, North',
+					ne: 'Nepali',
+					ng: 'Ndonga',
+					nl: 'Dutch',
+					nn: 'Norwegian Nynorsk',
+					no: 'Norwegian',
+					nr: 'Ndebele, South',
+					nv: 'Navajo',
+					ny: 'Nyanja',
+					oc: 'Occitan',
+					oj: 'Ojibwa',
+					om: 'Oromo',
+					or: 'Oriya',
+					os: 'Ossetian',
+					pa: 'Panjabi',
+					pi: 'Pali',
+					pl: 'Polish',
+					ps: 'Pushto',
+					pt: 'Portuguese',
+					qu: 'Quechua',
+					rm: 'Romansh',
+					rn: 'Rundi',
+					ro: 'Romanian',
+					ru: 'Russian',
+					rw: 'Kinyarwanda',
+					sa: 'Sanskrit',
+					sc: 'Sardinian',
+					sd: 'Sindhi',
+					se: 'Northern Sami',
+					sg: 'Sango',
+					si: 'Sinhala',
+					sk: 'Slovak',
+					sl: 'Slovenian',
+					sm: 'Samoan',
+					sn: 'Shona',
+					so: 'Somali',
+					sq: 'Albanian',
+					sr: 'Serbian',
+					ss: 'Swati',
+					st: 'Sotho',
+					su: 'Sundanese',
+					sv: 'Swedish',
+					sw: 'Swahili',
+					ta: 'Tamil',
+					te: 'Telugu',
+					tg: 'Tajik',
+					th: 'Thai',
+					ti: 'Tigrinya',
+					tk: 'Turkmen',
+					tl: 'Tagalog',
+					tn: 'Tswana',
+					to: 'Tonga',
+					tr: 'Turkish',
+					ts: 'Tsonga',
+					tt: 'Tatar',
+					tw: 'Twi',
+					ty: 'Tahitian',
+					ug: 'Uighur',
+					uk: 'Ukrainian',
+					ur: 'Urdu',
+					uz: 'Uzbek',
+					ve: 'Venda',
+					vi: 'Vietnamese',
+					vo: 'Volapük',
+					wa: 'Walloon',
+					wo: 'Wolof',
+					xh: 'Xhosa',
+					yi: 'Yiddish',
+					yo: 'Yoruba',
+					za: 'Zhuang',
+					zh: 'Chinese',
+					zu: 'Zulu',
```

Comment:
This most likely already exists, but I can see where maintaining this list could be an issue.
We already maintain such list here: https://github.com/nextcloud/server/blob/9de329a4c2327767d86bd7f594b232eb56af0d01/resources/locales.json


We have a dedicated php method to get those ressources.
I'll suggest we use that instead (you can either have this injected in the page from a InitialState or maybe it's already available
https://github.com/nextcloud/server/blob/9de329a4c2327767d86bd7f594b232eb56af0d01/lib/public/L10N/IFactory.php#L88

Reply:
Thanks @skjnldsv... so I would like to invoke L10N's php function 'findLanguageFromLocale()' from 'fetchTracks()'.
Please would you provide guidance as to how to import this function into Videos.vue?

public function findLanguageFromLocale(string $app = 'core', string $locale = null)

Reply:
Why do you need the findLanguageFromLocale ?
All of the locale computation should be done on the php side